source/Preprocessor.cpp

```diff
@@ -395,28 +396,70 @@ Trivia Preprocessor::handleEndIfDirective(Token directive) {
     return parseBranchDirective(directive, Token(), taken);
 }
 
-void Preprocessor::expectTimescaleSpecifier(Token& value, Token& unit) {
-    // TODO: check for allowed values
-    auto token = peek();
-    if (token.kind == TokenKind::IntegerLiteral) {
+bool Preprocessor::expectTimescaleSpecifier(Token& value, Token& unit) {
+    if (peek().kind == TokenKind::IntegerLiteral) {
         value = consume();
         unit = expect(TokenKind::Identifier);
+    } else {
+        auto token = expect(TokenKind::TimeLiteral);
```

Comment:
Rather than expect() here, maybe just use peek() and look for the TimeLiteral token. That way we don't potentially create a new missing token with some valid trivia (like if there was whitespace there or something) and then throw it away.

Basically if peek() doesn't give us an integer literal or a time literal, just expect(IntegerLiteral) expect(Identifier) and we're done.